cmd/api/src/database/saved_queries_permissions.go

```diff
@@ -52,16 +69,77 @@ func (s *BloodhoundDB) CreateSavedQueryPermissionToPublic(ctx context.Context, q
 	return permission, CheckError(s.db.WithContext(ctx).Create(&permission))
 }
 
+// CreateSavedQueryPermissionsBatch attempts to save the given saved query permissions in batches of 100 in a transaction
+func (s *BloodhoundDB) CreateSavedQueryPermissionsBatch(ctx context.Context, savedQueryPermissions []model.SavedQueriesPermissions) ([]model.SavedQueriesPermissions, error) {
+	result := s.db.WithContext(ctx).Clauses(clause.OnConflict{
+		DoNothing: true,
```

Comment:
Since we are `on conflict, do nothing` do we actually even need to do the `hasAccess` db check above? 🤔 @irshadaj 

If we do keep the check, we might consider leveraging the `GetPermissionsForSavedQuery` and checking access via logic vs hitting the db for each shared user supplied.